pkg/graveler/committed/meta_range.go

```diff
@@ -24,6 +24,29 @@ type Iterator interface {
 	Close()
 }
 
+// DiffIterator iterates over all Range headers and values of a Diff, allowing seeking by entire
+// ranges.
+type DiffIterator interface {
+	// Next moves to look at the next value in the current Range, or a header for the next Range if the current Range is over and a next range exists.
+	Next() bool
+	// NextRange skips the current range
+	// possible only if we are currently inside a range
+	// the next value could be Range or Value
```

Comment:
Why?

Reply:
We could only get a range when one of the ranges is completely before the other range,
In case we have overlapping ranges we could not return range.